mobile/asa-go/src/components/map/ASAGoMap.tsx

```diff
@@ -269,13 +279,75 @@ const ASAGoMap = (props: ASAGoMapProps) => {
       }),
       interactions: defaultInteractions({
         doubleClickZoom: true,
-      }).extend([new DblClickDragZoom()]),
+      }),
     });
     mapObject.setTarget(mapRef.current);
 
+    /******* Start double click drag zoom interaction ******/
+
+    const dblClickDragZoom = new DblClickDragZoom();
+
+    const originalHandleDownEvent =
+      dblClickDragZoom.handleDownEvent.bind(dblClickDragZoom);
+    dblClickDragZoom.handleDownEvent = (event) => {
+      doubleClkDragging.current = true;
+      return originalHandleDownEvent(event);
+    };
+
+    const originalHandleUpEvent =
+      dblClickDragZoom.handleUpEvent.bind(dblClickDragZoom);
+    dblClickDragZoom.handleUpEvent = (event) => {
+      doubleClkDragging.current = false;
+      return originalHandleUpEvent(event);
+    };
+    mapObject.addInteraction(dblClickDragZoom);
+
+    /******* End double click drag zoom interaction ******/
+
+    /******* Start scale line ******/
+
     const scaleBar = new ScaleLine({});
     scaleBar.setTarget(scaleRef.current);
     mapObject.addControl(scaleBar);
+    const setScalelineVisibility = () => {
+      setScaleVisible(true);
+    };
+
+    // Make the scale line visible when the user zooms in/out
+    mapObject.getView().on("change:resolution", setScalelineVisibility);
+
+    /******* End scale line ******/
+
+    /******* Start map popup ******/
+    popup.setElement(popupRef.current);
+    mapObject.addOverlay(popup);
+    const mapClickHandler = (event: MapBrowserEvent<UIEvent>) => {
+      if (doubleClkDragging.current) {
+        return;
```

Comment:
Can we add a test to make sure we handle the `doubleClkDragging` state?

Reply:
I'm pulling the double click and drag to zoom functionality for the skateboard.